eng/pipelines/repoman-gha.yml

```diff
@@ -0,0 +1,173 @@
+          and(
+            succeeded(),
+            or(
+              eq(variables['BuildReasonOverride'], 'PullRequest'),
+              and(
+                eq('', variables['BuildReasonOverride']),
+                eq(variables['Build.Reason'], 'PullRequest')
+              )
+            )
+          )
+        pool:
+          name: azsdk-pool-mms-ubuntu-2004-general
+          vmImage: MMSUbuntu20.04
+
+        variables:
+          RepoManResultsFile: $(Pipeline.Workspace)/repoman.md
+
+        steps:
+          - checkout: self
+
+          - pwsh: |
+              $PRNumber = '$(System.PullRequest.PullRequestNumber)'
+              if ($env:PRNUMBEROVERRIDE) {
+                Write-Host "PR Number override: $($env:PRNUMBEROVERRIDE)"
+                $PRNumber = "$($env:PRNUMBEROVERRIDE)"
+              }
+              Write-Host "##vso[task.setvariable variable=PRNumber]$PRNumber"
+            displayName: Set PRNumber
+
+          - template: /eng/pipelines/templates/steps/repoman-generate.yml
+            parameters:
+              TargetBranchName: pr/$(PRNumber)
+              RemoteName: $(StagingRemoteName)
+              ResultsFileLocation: $(RepoManResultsFile)
+
+          - task: PowerShell@2
+            name: DetermineChanged
+            displayName: Set repoman changed templates
+            inputs:
+              pwsh: true
+              targetType: filePath
+              filePath: ./eng/scripts/Set-RepomanChangedRepositories.ps1
+              arguments: >-
+                -ResultsFile $(RepoManResultsFile)
+                -OutputTemplatesVariable "TemplateList"
+                -OutputTemplateBranchVariable "TemplateBranch"
+
+          - pwsh: |
+              $repomanContent = "No changes detected."
+              $repomanOutputFile = "$(RepoManResultsFile)"
+              if (Test-Path $repomanOutputFile) {
+                $repomanContent = Get-Content $repomanOutputFile -Raw
+              }
+
+              $tag ='<!-- #comment-repoman-generate -->'
+              $content = @"
+              $tag
+              ## Repoman Generation Results
+              Repoman pushed changes to remotes for the following projects:
+              $repomanContent
+              "@
+              $file = New-TemporaryFile
+              Set-Content -Path $file -Value $content
+              Write-Host "##vso[task.setvariable variable=CommentBodyFile]$file"
+            displayName: Set Repoman Content
+
+          - task: PowerShell@2
+            displayName: Post PR comment
+            inputs:
+              pwsh: true
+              targetType: filePath
+              filePath: ./eng/scripts/Update-PRComment.ps1
+              arguments: >-
+                -Repo 'azure/azure-dev'
+                -PRNumber '$(PRNumber)'
+                -BodyFile '$(CommentBodyFile)'
+                -Tag '<!-- #comment-repoman-generate -->'
+            env:
+              GH_TOKEN: $(azuresdk-github-pat)
```

Comment:
We need to ask @weshaggard  or @danieljurek to add this secret to this new Pipeline.

Reply:
that's why you're getting: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3333420&view=logs&j=dc5beac1-197e-530f-c839-b08cb3c610f0&t=97b95599-a840-5037-d7c8-193d553558d7&l=14

Reply:
Can I have some more context on what this pipeline is for? Is it important for any PR to be able to trigger it or only on PRs from team members or ones team members trigger this explicitly on?

Reply:
> what this pipeline is for?

We are moving the code from https://github.com/Azure/setup-azd to azure-dev repo.  We will use setup-azd just to host the gh-action /dist folder and top level files (readme, notice, action.yaml, etc).

We are starting by moving the entire repo and having repo-man to sync whenever we make any changes.

> only on PRs from team members or ones team members trigger this explicitly on?

Yes, only from members (forks as well). We want the same exact model we have for the sample-templates, but we are not using the existing pipeline because it has specific steps for the samples (runs tests and more stuff unique for samples).